sklearn/metrics/tests/test_common.py

```diff
@@ -638,20 +640,27 @@ def test_symmetric_metric(name):
 
 @pytest.mark.parametrize("name", sorted(NOT_SYMMETRIC_METRICS))
 def test_not_symmetric_metric(name):
+
     # Test the symmetry of score and loss functions
     random_state = check_random_state(0)
-    y_true = random_state.randint(0, 2, size=(20,))
-    y_pred = random_state.randint(0, 2, size=(20,))
+    metric = ALL_METRICS[name]
 
-    if name in METRICS_REQUIRE_POSITIVE_Y:
-        y_true, y_pred = _require_positive_targets(y_true, y_pred)
+    always_symmetric = True
+    for _ in range(5):
+        y_true = random_state.randint(0, 2, size=(20,))
+        y_pred = random_state.randint(0, 2, size=(20,))
 
-    metric = ALL_METRICS[name]
+        if name in METRICS_REQUIRE_POSITIVE_Y:
+            y_true, y_pred = _require_positive_targets(y_true, y_pred)
 
-    # use context manager to supply custom error message
-    with pytest.raises(AssertionError):
-        assert_array_equal(metric(y_true, y_pred), metric(y_pred, y_true))
-        raise ValueError("%s seems to be symmetric" % name)
+        nominal = metric(y_true, y_pred)
+        swapped = metric(y_pred, y_true)
+        if not np.allclose(nominal, swapped):
+            always_symmetric = False
+            break
+
+    if always_symmetric:  # pragma: no cover
+        raise ValueError(f"{name} seems to be symmetric")
```

Comment:
```suggestion
    if not always_symmetric:
        raise ValueError(f"{name} seems to be asymmetric")
```
There should be a test for this, e.g. applying `test_not_symmetric_metric` to log_loss and test for a fail.

Reply:
The meta test `test_symmetry_tests` in 6de5e13 checks `test_symmetric_metric` and `test_not_symmetric_metric`.